src/masternode.cpp

```diff
@@ -557,10 +557,9 @@ bool CMasternodeBroadcast::CheckInputsAndAdd(int& nDos)
     CMasternode mn(*this);
     mnodeman.Add(mn);
 
-    bool isLocal = addr.IsRFC1918() || addr.IsLocal();
-    if(Params().NetworkIDString() == CBaseChainParams::REGTEST) isLocal = false;
-
-    if(!isLocal) Relay();
+    bool isLocal = (addr.IsRFC1918() || addr.IsLocal());
+    // Not Relay for local address on Main network
+    if(!(isLocal && Params().NetworkIDString() == CBaseChainParams::MAIN)) Relay();
```

Comment:
wouldn't it be a bit easier to read if this was `if(isLocal && Params().NetworkIDString() == CBaseChainParams::MAIN) return true;` instead and then `Relay();` on the next line?